File: backend/app/models/category.py
import uuid
from sqlalchemy import Column, String, Integer, ForeignKey
from sqlalchemy.dialects.postgresql import UUID
from app.db.base import Base

class Category(Base):
    __tablename__ = "categories"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    user_id = Column(UUID(as_uuid=True), ForeignKey("users.id"), nullable=False)
    name = Column(String, nullable=False)
    priority_points = Column(Integer, nullable=False)
    description = Column(String, nullable=True)

File: frontend/lib/models/category.dart
class Category {
  final String id;
  final String name;
  final int priorityPoints;
  final String? description;

  Category({
    required this.id,
    required this.name,
    required this.priorityPoints,
    this.description,
  });

  factory Category.fromJson(Map<String, dynamic> json) {
    return Category(
      id: json['id'],
      name: json['name'],
      priorityPoints: json['priority_points'],
      description: json['description'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'priority_points': priorityPoints,
      'description': description,
    };
  }
}
